fenix/app/src/main/java/org/mozilla/fenix/debugsettings/tabs/TabManager.kt

```diff
@@ -0,0 +1,226 @@
+            inactiveTabCount = inactiveTabCount,
+            privateTabCount = privateTabCount,
+            totalTabCount = tabs.size,
+            inactiveTabsEnabled = inactiveTabsEnabled,
+        )
+    }
+}
+
+@Composable
+private fun TabCounter(
+    normalTabCount: Int,
+    inactiveTabCount: Int,
+    privateTabCount: Int,
+    totalTabCount: Int,
+    inactiveTabsEnabled: Boolean,
+) {
+    Column {
+        Text(
+            text = stringResource(R.string.debug_drawer_tab_manager_tab_count_title),
+            color = FirefoxTheme.colors.textPrimary,
+            style = FirefoxTheme.typography.headline5,
+        )
+
+        Spacer(modifier = Modifier.height(16.dp))
+
+        TabCountRow(
+            text = stringResource(R.string.debug_drawer_tab_manager_tab_count_normal),
+            count = normalTabCount.toString(),
+        )
+
+        TabCountRow(
+            text = stringResource(R.string.debug_drawer_tab_manager_tab_count_inactive),
+            count = if (inactiveTabsEnabled) {
+                inactiveTabCount.toString()
+            } else {
+                "0*"
+            },
+        )
+
+        TabCountRow(
+            text = stringResource(R.string.debug_drawer_tab_manager_tab_count_private),
+            count = privateTabCount.toString(),
+        )
+
+        Spacer(modifier = Modifier.height(8.dp))
+
+        Divider()
+
+        Spacer(modifier = Modifier.height(8.dp))
+
+        TabCountRow(
+            text = stringResource(R.string.debug_drawer_tab_manager_tab_count_total),
+            count = totalTabCount.toString(),
+        )
+
+        if (!inactiveTabsEnabled) {
+            Text(
+                text = stringResource(R.string.debug_drawer_tab_manager_inactive_tabs_disabled),
+                modifier = Modifier.align(Alignment.End),
+                color = FirefoxTheme.colors.textPrimary,
+                style = FirefoxTheme.typography.caption,
+            )
+        }
+    }
+}
+
+@Composable
+private fun TabCountRow(
+    text: String,
+    count: String,
+) {
+    Row(
+        modifier = Modifier
+            .fillMaxWidth()
+            .padding(start = 16.dp),
+        horizontalArrangement = Arrangement.SpaceBetween,
+    ) {
+        Text(
+            text = text,
+            color = FirefoxTheme.colors.textSecondary,
+            style = FirefoxTheme.typography.headline6,
+        )
+
+        Text(
+            text = count,
+            color = FirefoxTheme.colors.textSecondary,
+            style = FirefoxTheme.typography.headline6,
+        )
+    }
+}
+
+private data class TabManagerPreviewModel(
+    val normalTabs: List<TabSessionState> = emptyList(),
+    val inactiveTabs: List<TabSessionState> = emptyList(),
+    val privateTabs: List<TabSessionState> = emptyList(),
+    val inactiveTabsEnabled: Boolean = true,
+)
+
+private class TabManagerPreviewModelParameterProvider : PreviewParameterProvider<TabManagerPreviewModel> {
```

Comment:
I'm wondering if in the future (when this is in a module and we can use `internal`) if it would make sense to group all the preview utilities into a separate file. I think it could help de-clutter the Composable files.

Reply:
That would definitely be a boon. The preview parameters are super helpful but they inherently add some boilerplate bloat to the files.